### PR TITLE
Add UserTags support to list_objects API

### PR DESCRIPTION
**Pull request by angrybat**
This adds the tags property to the Object class which can be set in the constructor of the class. This defaults to an empty Tags class. 

The http response already contains the tags in the xml response under the text in the "UserTags" key. The fromxml method has been modified to set the tags to a populated version of the Tags class. This used the added Tags.object_from_user_tags_text method.

The Tags.object_from_user_tags_text method is a new method that creates a Tags class instance for Objects from a string in the format "key1=value1&key2=value2". This is the format the tags are in the text of the "UserTags" key.

### CODE REVIEW

**Review by balamurugana — CHANGES_REQUESTED** [review]
Filtering objects by any parameter should be done outside of `minio-py`. Please keep the PR focusing on adding `UserTags` support.

**Comment by angrybat** [comment]
@balamurugana out of interest how come filtering by UserTags should be done outside of minio-py? 

The list_objects function has a prefix parameter that allows you to filter by path. How come that is different to filtering by UserTags?

The filtering I have added here is optional. So the original functionality of the function is preserved.

**Comment by balamurugana** [comment]
> @balamurugana out of interest how come filtering by UserTags should be done outside of minio-py?
> 
> The list_objects function has a prefix parameter that allows you to filter by path. How come that is different to filtering by UserTags?
> 
> The filtering I have added here is optional. So the original functionality of the function is preserved.

Filtering by `prefix` and other parameters are done at server side whereas `UserTags` filtering is done at client side in this PR which we don't want.

File: minio/datatypes.py
Comment (balamurugana) [inline minio/datatypes.py:136]: It should be `tags: Tags | None = None`

File: minio/commonconfig.py
Comment (balamurugana) [inline minio/commonconfig.py:87]: Remove this method and move the logic into https://github.com/minio/minio-py/pull/1381/files#diff-321a5d1087d9a70eb00a39c6bfe8b7295ff787da4efe97f7fe7fd16aed5c2df3R265

**Review by balamurugana — CHANGES_REQUESTED** [review]

File: minio/datatypes.py
Comment (balamurugana) [inline minio/datatypes.py:269]:
```suggestion
        tags: Tags | None = None
        if tag_text:
            tags = Tags.new_object_tags()
            tags.update([tokens.split("=") for tokens in tag_text.split("&")])
```

**Review by balamurugana — CHANGES_REQUESTED** [review]

**Review by balamurugana — DISMISSED** [review]

File: minio/datatypes.py
Comment (balamurugana) [inline minio/datatypes.py:272]:
```suggestion
            tags.update([tokens.split("=") for tokens in tags_text.split("&")])
```

File: minio/datatypes.py
Comment (balamurugana) [inline minio/datatypes.py:270]:
```suggestion
        if tags_text:
```

File: minio/datatypes.py
Comment (balamurugana) [inline minio/datatypes.py:272]:
```suggestion
            tags.update(
                [tuple(tokens.split("=")) for tokens in tags_text.split("&")],
            )
```

File: minio/datatypes.py
Comment (balamurugana) [inline minio/datatypes.py:273]:
```suggestion
                cast(
                    List[Tuple[Any, Any]],
                    [tokens.split("=") for tokens in tags_text.split("&")],
                ),
```

**Review by balamurugana — CHANGES_REQUESTED** [review]
import `List`, `Tuple`, `Any` at line no. 30

**Review by balamurugana — APPROVED** [review]

**Comment by angrybat** [comment]
Can I get a second review please

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**
